core/auth/v2/authenticator.go

```diff
@@ -32,24 +30,16 @@ func (*authenticator) AuthenticateBlobRequest(header *core.BlobHeader) error {
 		return fmt.Errorf("failed to get blob key: %v", err)
 	}
 
-	publicKeyBytes, err := hexutil.Decode(header.PaymentMetadata.AccountID)
-	if err != nil {
-		return fmt.Errorf("failed to decode public key (%v): %v", header.PaymentMetadata.AccountID, err)
-	}
-
-	// Decode public key
-	pubKey, err := crypto.UnmarshalPubkey(publicKeyBytes)
-	if err != nil {
-		return fmt.Errorf("failed to convert bytes to public key (%v): %v", header.PaymentMetadata.AccountID, err)
-	}
-
-	// Verify the signature
+	// Recover public key from signature
 	sigPublicKeyECDSA, err := crypto.SigToPub(blobKey[:], sig)
 	if err != nil {
 		return fmt.Errorf("failed to recover public key from signature: %v", err)
 	}
 
-	if !bytes.Equal(pubKey.X.Bytes(), sigPublicKeyECDSA.X.Bytes()) || !bytes.Equal(pubKey.Y.Bytes(), sigPublicKeyECDSA.Y.Bytes()) {
+	accountId := header.PaymentMetadata.AccountID
+	pubKey := crypto.PubkeyToAddress(*sigPublicKeyECDSA).Hex()
+
+	if pubKey != accountId {
```

Comment:
Here we're comparing strings, but this will fail if `accountID` isn't checksummed string. 
I think it's safer to convert `accountID` to common.Address (common.HexToAddress()) and compare bytearrays using `Address.Cmp()`

Reply:
Updated to use address type.

We have auth_test that check for authenticating the requests, but not as specific as whether it is checksum or not.